pkgs/by-name/br/breakpointHook/attach.sh

```diff
@@ -0,0 +1,28 @@
+#! /usr/bin/env bash
+
+id="$1"
+pid="$(pgrep -f "sleep $id" | head -n1)"
+
+# helper to extract variables from the build env
+getVar(){
+  nsenter --mount --target "$pid" /bin/sh -c "
+    while read line; do
+      case \$line in
+        *\"declare -x $1=\"* )
+          echo \$line
+            ;;
+      esac
+    done < /build/env-vars
+  " | cut -d \" -f 2
+}
+
+# the build shell is needed to `source /build/env-vars``
+buildShell="$(getVar shell)"
+# the debug shell will be started as interactive shell after sourcing /build/env-vars
+debugShell="$(getVar debugShell)"
+# to drop the user into the working directory at the point of failure
+pwd="$(getVar PWD)"
+
+# enter the namespace of the failed build
+exec nsenter --mount --target "$pid" "$buildShell" -c \
```

Comment:
We might not need `buildShell` if we can use `environ`. It should work with `$debugShell`. Also there are potential derivations that uses a different builder... also those don't support this build hook potentially.

Reply:
Potentially there are shells that don't support the `export` syntax? like some python based stuff.

Reply:
Alright, since I'm depending on bashInteractive anyways, due to it being the default interactive shell, I can just use `bashInteractive` to source the env